commit ee76b399a8cb19702dde8834deb47c237d33f7bc (refs/changes/23/3923/2)
Author: Dave Eddy <dave@daveeddy.com>
Date:   2018-05-09T17:23:24-04:00 (1 year, 5 months ago)
    
    OS-6942 bhyve incapable hosts should exit early on vmadm create
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index d82ab6d8..416316d5 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -10036,6 +10036,35 @@ exports.create = function (payload, options, callback)
         }, function (cb) {
             var vm_type = BRAND_OPTIONS[payload.brand].features.type;
 
+            switch (vm_type) {
+            case 'BHYVE':
+                VM.getSysinfo(function gotSysinfo(err, sysinfo) {
+                    if (err) {
+                        cb(err);
+                        return;
+                    }
+
+                    if (!sysinfo.hasOwnProperty('Bhyve Capable')) {
+                        cb(new Error(
+                            'sysinfo does not have "Bhyve Capable" property'));
+                        return;
+                    }
+
+                    if (!sysinfo['Bhyve Capable']) {
+                        cb(new Error('Bhyve not supported'));
+                        return;
+                    }
+
+                    cb();
+                });
+                break;
+            default:
+                cb();
+                break;
+            }
+        }, function (cb) {
+            var vm_type = BRAND_OPTIONS[payload.brand].features.type;
+
             if (['BHYVE', 'KVM'].indexOf(vm_type) !== -1) {
                 createVM(payload, log, function (error, result) {
                     if (error) {
